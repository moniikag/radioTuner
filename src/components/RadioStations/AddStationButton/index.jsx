import React from 'react'
import PropTypes from 'prop-types'

const AddStationButton = ({ handleClick }) => {

  return (
    <li>
      <button
        className="add"
        onClick={handleClick}
      >
        +
      </button>
    </li>
  )
}

AddStationButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
}

export default AddStationButton
