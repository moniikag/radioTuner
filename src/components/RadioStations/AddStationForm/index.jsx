import React from 'react'
import PropTypes from 'prop-types'

const AddStationForm = ({ handleClose }) => {

  return (
    <li id="form">
      <button
        id="close"
        onClick={handleClose}
      >
        x
      </button>
      <form>
        <input id="name" type="text" placeholder="name" />
        <input type="text" placeholder="frequency" />
      </form>
    </li>
  )
}

AddStationForm.propTypes = {
  handleClose: PropTypes.func.isRequired,
}

export default AddStationForm
