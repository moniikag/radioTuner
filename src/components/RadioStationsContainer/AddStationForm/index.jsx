import React, { Component } from 'react'
import PropTypes from 'prop-types'

class AddStationForm extends Component {

  constructor(props) {
    super(props)
    this.state = {}
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(value, field) {
    const newStation = { ...this.state.newStation, [field]: value }
    this.setState({ newStation })
  }

  handleSubmit(e) {
    e.preventDefault()
    const { newStation } = this.state
    if(newStation) { this.props.handleSubmit(newStation) }
    this.props.handleClose()
  }

  render() {
    return (
      <li id="form">
        <button
          id="close"
          onClick={this.props.handleClose}
        >
          x
        </button>
        <form onSubmit={(e) => this.handleSubmit(e)}>
          <input
            id="name"
            onChange={(e) => this.handleChange(e.target.value, 'name')}
            type="text"
            placeholder="name"
          />
          <input
            onChange={(e) => this.handleChange(e.target.value, 'frequency')}
            type="text"
            placeholder="frequency"
          />
          <button style={{display: 'none'}} type="submit"></button>
        </form>
      </li>
    )
  }
}

AddStationForm.propTypes = {
  handleClose: PropTypes.func.isRequired,
}

export default AddStationForm
