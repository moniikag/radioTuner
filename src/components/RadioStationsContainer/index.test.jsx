import React from 'react'
import { shallow, mount } from 'enzyme'

import { RadioStations } from './index'
import { MemoryRouter } from 'react-router-dom'
import AddStationButton from './AddStationButton'
import AddStationForm from './AddStationForm'


describe('RadioStations', () => {
  describe('content', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<RadioStations/>)
    })

    it('renders as a article', () => {
      expect(wrapper.is('article')).toBe(true)
    })

    it('renders add station button by default', () => {
      expect(wrapper.find(AddStationButton).length).toBe(1)
    })

    it('does not render new station form by default', () => {
      expect(wrapper.find(AddStationForm).length).toBe(0)
    })

    it('renders form instead of button for the relevant state', () => {
      wrapper.setState({ addItemFormOpen: true })
      expect(wrapper.find(AddStationButton).length).toBe(0)
      expect(wrapper.find(AddStationForm).length).toBe(1)
    })
  })

  describe('behaviour', () => {
    const props = { addRadioStation: jest.fn() }
    const wrapper = mount(
      <MemoryRouter>
        <RadioStations {...props}/>
      </MemoryRouter>
    )

    it('updates component state on new station button click', () => {
      expect(wrapper.find(AddStationForm).length).toBe(0)
      const button = wrapper.find(AddStationButton).find('button')
      button.simulate('click')
      expect(wrapper.find(AddStationForm).length).toBe(1)
    })

    it('calls addRadioStation and passes correct params', () => {
      const e1 = { target: { value: 'RadioWaw' } }
      const e2 = { target: { value: '99.1' } }
      wrapper.find('form').find('input').at(0).simulate('change', e1)
      wrapper.find('form').find('input').at(1).simulate('change', e2)
      wrapper.find('form').simulate('submit', {})
      const call = props.addRadioStation.mock.calls[0][0]
      expect(call.name).toEqual('RadioWaw')
      expect(call.frequency).toEqual('99.1')
      expect(call.id).not.toBe(undefined)
      expect(call.id.length).toBe(36)
    })
  })
})
