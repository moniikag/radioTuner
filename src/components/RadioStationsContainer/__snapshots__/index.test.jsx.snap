// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioStations snapshots matches mount snapshot 1`] = `
<RadioStations
  stations={Array []}
>
  <article
    className="main-article"
  >
    <section
      className="main-section"
    >
      <ul
        className="radio-stations-list"
      >
        <AddStationForm
          handleClose={[Function]}
          handleSubmit={[Function]}
        >
          <li
            id="form"
          >
            <button
              id="close"
              onClick={[Function]}
            >
              x
            </button>
            <form
              onSubmit={[Function]}
            >
              <input
                id="name"
                onChange={[Function]}
                placeholder="name"
                type="text"
              />
              <input
                onChange={[Function]}
                placeholder="frequency"
                type="text"
              />
              <button
                style={
                  Object {
                    "visibility": "hidden",
                  }
                }
                type="submit"
              />
            </form>
          </li>
        </AddStationForm>
      </ul>
    </section>
    <Route
      component={[Function]}
      path="/stations/:stationId"
    />
  </article>
</RadioStations>
`;

exports[`RadioStations snapshots matches render snapshot 1`] = `
<article
  class="main-article"
>
  <section
    class="main-section"
  >
    <ul
      class="radio-stations-list"
    >
      <li>
        <button
          class="delete-station"
        >
          x
        </button>
        <a
          aria-current="false"
          class="station-link"
          href="/stations/undefined"
        >
          <header>
            X
          </header>
        </a>
        Frequency:
        <span
          class="station-detail"
        >
          121.1 FM
        </span>
      </li>
      <li>
        <button
          id="add"
        >
          +
        </button>
      </li>
    </ul>
  </section>
</article>
`;

exports[`RadioStations snapshots matches shallow snapshot 1`] = `
<article
  className="main-article"
>
  <section
    className="main-section"
  >
    <ul
      className="radio-stations-list"
    >
      <AddStationButton
        handleClick={[Function]}
      />
    </ul>
  </section>
  <Route
    component={[Function]}
    path="/stations/:stationId"
  />
</article>
`;
